src/container/BooksList.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Book from '../components/Book';
+
+function BooksList(props) {
+  const { books } = props;
+
+  return (
+    <table>
+      <tr>
+        <th>Book ID</th>
+        <th>Title</th>
+        <th>Category</th>
+      </tr>
+
+      { books.map((book) => (<Book key={book.id} book={book} />)) }
+
+    </table>
+  );
+}
+
+BooksList.propTypes = {
+  books: PropTypes.arrayOf(PropTypes.shape({
```

Comment:
- [ ] As long as you initialize the books array in the store as an empty array you can mark this prop as required and not include a default fallback prop for it.